pyproject.toml

```diff
@@ -95,7 +95,7 @@ full = [
     "scikit-learn",
     "networkx",
     "distinctipy",
-    "matplotlib<3.9",  # See https://github.com/SpikeInterface/spikeinterface/issues/2863
+    "matplotlib",
```

Comment:
I don't think but might require a lower bound. Let me check.

Reply:
It seems that matplotlib.colormaps is 3.5 at least 
https://matplotlib.org/stable/api/prev_api_changes/api_changes_3.6.0.html#pending-deprecation-top-level-cmap-registration-and-access-functions-in-mpl-cm


And the resampler might requires lower bound of 3.6:

https://matplotlib.org/stable/users/prev_whats_new/whats_new_3.6.0.html#colormap-method-for-creating-a-different-lookup-table-size

I would set 3.6 as a minimal if you don't want to fiddle with it.

Reply:
Copying screenshot because this is not displayed correctly:
![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/c40ee56b-119b-4c3e-9bfe-c4e4fa65caa2)

Reply:
Another option is to use pyplot.get_cmap() which they still have for backwards compatbility and you don't need lower bound I think. 

But I think that a lower bound is a good idea at this point.